okhttp/src/test/java/io/grpc/transport/okhttp/OkHttpClientTransportTest.java

```diff
@@ -598,6 +599,50 @@ public void receivingWindowExceeded() throws Exception {
     verify(frameWriter).rstStream(eq(3), eq(ErrorCode.FLOW_CONTROL_ERROR));
   }
 
+  @Test
+  public void receiveDataWithoutHeader() throws Exception {
+    receiveOnlyDataShouldFail(true, false);
+  }
+
+  @Test
+  public void receiveDataWithoutHeaderAndTrailer() throws Exception {
+    receiveOnlyDataShouldFail(false, false);
+    receiveOnlyDataShouldFail(false, true);
+  }
+
+  private void receiveOnlyDataShouldFail(boolean hasTrailer, boolean longEnoughMessage)
+      throws Exception {
+    MockStreamListener listener = new MockStreamListener();
+    clientTransport.newStream(method,new Metadata.Headers(), listener).request(1);
+    assertEquals(1, streams.size());
+    int streamId = streams.values().iterator().next().id();
+    Buffer buffer = createMessageFrame(new byte[longEnoughMessage ? 1000 : 1]);
+    int messageFrameLength = (int) buffer.size();
+    frameHandler.data(false, streamId, buffer, messageFrameLength);
+
+    if (!longEnoughMessage) {
```

Comment:
These conditions don't make me feel good. Can we just copy/paste the code between the tests? It looks like if we did that we wouldn't need "int streamId", we wouldn't need messageFrameLength, and we wouldn't need the conditions. The boilderplate at that point would be quite readable and doesn't look problematic to duplicated.